doc/api_assets/api.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+// Check if we have JavaScript support
+document.querySelector(':root').classList.add('has-js');
+
+// Restore user mode preferences
+const kCustomPreference = 'customDarkTheme';
+const userSettings = sessionStorage.getItem(kCustomPreference);
+const themeToggleButton = document.getElementById('theme-toggle-btn');
```

Comment:
It's unfortunate to create global variables here. Can we make a scope? (Or maybe add `type="module"` to the script tag?)

Reply:
Yes, I restructured the code to have three big function to clearly separate what we are doing.